src/components/editor-page/app-bar/app-bar.tsx

```diff
@@ -45,12 +48,22 @@ export const AppBar: React.FC<AppBarProps> = ({ mode }) => {
           <SyncScrollButtons/>
         </ShowIf>
         <DarkModeButton/>
-        <Link to={ `/p/${ id }` } target='_blank'>
-          <Button title={ t('editor.documentBar.slideMode') } className="ml-2 text-secondary" size="sm"
-                  variant="outline-light">
-            <ForkAwesomeIcon icon="television"/>
-          </Button>
-        </Link>
+        <ShowIf condition={noteFrontmatter.type === NoteType.SLIDE}>
+          <Link to={ `/p/${ id }` } target='_blank'>
+            <Button title={ t('editor.documentBar.slideMode') } className="ml-2 text-secondary" size="sm"
+                    variant="outline-light">
+              <ForkAwesomeIcon icon="television"/>
+            </Button>
+          </Link>
+        </ShowIf>
+        <ShowIf condition={noteFrontmatter.type !== NoteType.SLIDE}>
+          <Link to={ `/s/${ id }` } target='_blank'>
+            <Button title={ t('editor.documentBar.readOnlyMode') } className="ml-2 text-secondary" size="sm"
+                    variant="outline-light">
+              <ForkAwesomeIcon icon="file-text-o"/>
+            </Button>
+          </Link>
+        </ShowIf>
```

Comment:
This is getting a bit long. Can you extract the buttons into components? Like the ones below?

Reply:
Done